pkg/scyllaclient/client_rclone.go

```diff
@@ -460,22 +463,42 @@ func (c *Client) RcloneListDirIter(ctx context.Context, host, remotePath string,
 		}
 	}
 
+	more := make(chan bool)
+	errCh := make(chan error)
+
 	var v RcloneListDirItem
-	for dec.More() {
-		// Detect context cancellation
-		if ctx.Err() != nil {
-			return ctx.Err()
+	go func() {
```

Comment:
It looks like we cannot do without a go routine, but we shall clean it up.

The reading go routine should take a token form channel and return result to channel, on time out the token channel is closed. In other words one channel in one channel out.

Reply:
Sounds good will clean up. Could you explain though what you mean by "token form channel"?

Reply:
Let me elaborate a little bit...

I think that we could have a single output channel from the go routine.
It technically can be the error channel:
* nil error - OK
* not nil error - NOK
* closed - Done

Also, there need to be a way to inform the go routine to stop.
This is only relevant when it is locked on More() refilling the data...

When it times out we should close the body making the More() return false.

Reply:
Ergo the input channel may not exist.

Reply:
Thanks - implemented like this and it works great and is cleaner.

I did not close the body explicitly when it times out since that is deferred earlier on and we are returning with an error when there is a timeout.